controllers: create kata-se runtime class

ajaypvictor opened the pull request:
Added IBM SE support in RuntimeClass handler as kata-se. Refactored multiple TEE detection to use a counter for presence checks instead of hardcoded pairwise conditions.

Closes: [KATA-4177](https://issues.redhat.com//browse/KATA-4177)

beraldoleal commented on controllers/confidential_handler.go:
```diff
@@ -18,16 +18,18 @@ const (
 //
 // For peer pods: sets ImageConfigMap and peer pods configMap to enable confidential images and CVM support.
-// For baremetal: creates kata-cc runtime classes with TEE-specific handlers (Intel TDX or AMD SNP).
+// For baremetal: creates kata-cc runtime classes with TEE-specific handlers (Intel TDX or AMD SNP or IBM SE).
```
nit: "Intel TDX, AMD, SNP or IBM SE"

ajaypvictor replied:
thanks beraldo, updated this..

beraldoleal commented on controllers/confidential_handler.go:
```diff
@@ -104,7 +106,7 @@ func (r *KataConfigOpenShiftReconciler) handleConfidentialPeerPods(state Feature
 
 // handleConfidentialBaremetal configures confidential computing for baremetal deployments.
-// It manages kata-cc runtime classes with TEE-specific handlers (Intel TDX or AMD SNP).
+// It manages kata-cc runtime classes with TEE-specific handlers (Intel TDX or AMD SNP or IBM SE).
```
same here.

ajaypvictor replied:
incorporated..

beraldoleal commented on controllers/confidential_handler.go:
```diff
@@ -177,6 +187,9 @@ func (r *KataConfigOpenShiftReconciler) computeTEEHandlerAndLabel() (string, str
+	}
 
-	return "", "", fmt.Errorf("no TEE platform labels found (expected %s or %s)", intelTDXNodeLabel, amdSNPNodeLabel)
+	return "", "", fmt.Errorf("no TEE platform labels found (expected %s or %s or %s)", intelTDXNodeLabel, amdSNPNodeLabel, ibmSENodeLabel)
```
same here.

ajaypvictor replied:
incorporated..

beraldoleal reviewed (commented):
Hi @ajaypvictor , thanks for this. LGMT. just a small nit comment about the ORs

beraldoleal commented on controllers/confidential_handler.go:
```diff
@@ -18,16 +18,18 @@ const (
 	// RuntimeClass handlers for TEE
 	kataCCIntelHandler = "kata-cc-intel"
 	kataCCAmdHandler   = "kata-cc-amd"
+	kataCCIbmHandler   = "kata-cc-ibm"
```
Vide, my PR #1229. We have agreed that we would be renaming the class handler... So maybe you would like to follow that convention as well?

Lets wait the discussion there, to see if makes sense to update here as well.

ajaypvictor replied:
sure, sounds good, thanks beraldo..

ajaypvictor replied:
updated the changes based on the https://github.com/openshift/sandboxed-containers-operator/pull/1229

ajaypvictor committed:
controllers: create kata-cc-ibm runtime class

Added IBM SE support in RuntimeClass handler as kata-cc-ibm. Refactored multiple TEE detection to use a counter for presence checks instead of hardcoded pairwise conditions.

Related: KATA-4177

Signed-off-by: Ajay Victor <ajvictor@in.ibm.com>

beraldoleal reviewed (approved):
LGTM, thanks.

gkurz reviewed (approved):
/lgtm

Thanks @ajaypvictor !

gkurz commented:
/ok-to-test

gkurz commented:
/retest
